caddy.go

```diff
@@ -497,14 +497,29 @@ func (d *Duration) UnmarshalJSON(b []byte) error {
 	var dur time.Duration
 	var err error
 	if b[0] == byte('"') && b[len(b)-1] == byte('"') {
-		dur, err = time.ParseDuration(strings.Trim(string(b), `"`))
+		dur, err = ParseDuration(strings.Trim(string(b), `"`))
 	} else {
 		err = json.Unmarshal(b, &dur)
 	}
 	*d = Duration(dur)
 	return err
 }
 
+// ParseDuration parses a duration string, adding
+// support for the "d" suffix for number of days.
+func ParseDuration(s string) (time.Duration, error) {
```

Comment:
This function will need tests (especially because it is so easy to test).

Reply:
I did add an adapt test which indirectly tests it, but yeah I'll add a `caddy_test.go` file I guess which does more direct ones

Reply:
I saw that -- thankful for that -- but we absolutely can unit test this sucker :)